### PR TITLE
patchedShaders in octEncodedNormals prevents redefinition

### PR DESCRIPTION
**Pull request by bkuster**
#189

### CODE REVIEW

**Comment by bkuster** [comment]
hold on, this still fails

**Comment by bkuster** [comment]
this is ready

**Comment by pjcozzi** [comment]
Thanks @bkuster!

@lilleyse could you look at this?

**Comment by lasalvavida** [comment]
@bkuster This looks good to me, thank you. Could you please add a test with two programs that reference the same shader?

Also, it doesn't need to be part of this PR, but it is worth mentioning that `compressTextureCoordinates` will have this issue as well.

File: lib/octEncodeNormals.js [hunk 1]
```diff
@@ -136,14 +137,18 @@ function patchTechnique(gltf, techniqueId) {
     }
 }
 
-function patchProgram(gltf, programId, normalVariableName) {
+function patchProgram(gltf, programId, normalVariableName, patchedShaders) {
     var programs = gltf.programs;
     var shaders = gltf.shaders;
     var program = programs[programId];
     var vertexShaderId = program.vertexShader;
+    if (patchedShaders[vertexShaderId]) {
```
Comment (lasalvavida) [inline lib/octEncodeNormals.js]: This should be `defined(patchedShaders[vertexShaderId])`. 

I know this works as-is, but `defined` is used throughout in this way to be more verbose.

**Comment by bkuster** [comment]
updated and wrote a _simple_ spec. I'll see what I can find in the `compressTextureCoordinates`

**Comment by bkuster** [comment]
I went ahead and just applied the same fix to the compressed coordinates. this is ready

**Comment by lasalvavida** [comment]
Yup, looks good to me, thanks @bkuster!

**Pull request merged**